src/qibolab/instruments/qm/config/pulses.py

```diff
@@ -24,6 +25,25 @@ def operation(pulse):
     return str(hash(pulse))
 
 
+def baked_duration(duration: int) -> int:
+    if duration < 16:
+        return 16
+    elif duration % 4 != 0:
+        return duration - (duration % 4) + 4
+    return duration
```

Comment:
```suggestion
    return np.maximum((duration + 3) // 4 * 4, 16)
```

In this way, it also works on arrays (and you need less branching).

However, working on arrays is not really needed, and you may find clearer to split the lower bound. But the other part I find clearer (since it's a single mathematical expression, against a conditional one).

Reply:
7e03145c7b60b98c3a7e13300b6411b6722a074f